Tidy up requests

westerlund opened the pull request:
I felt an urge to simplify and tidy up the existing code base, so I did!

Feel free to jam me with feedback if I did anything stupid.

Also, we really need tests for this project

westerlund committed:
Tidy up the requests and the base WebRequest

westerlund committed:
Shorten really long line

westerlund committed:
Remove unused parameter in callback

JagCesar reviewed (dismissed):
I just ran all our request against my car and all of them are working except `MobileEnabledForVehicleRequest`. But that isn't working on develop either, so Tesla has probably changed or killed this endpoint.

I'm slightly worried that we're moving creation of objects to the main thread. It isn't an issue atm (probably), and it makes the code easier to read.

So let's try it out, and if we run into trouble we'll fix it.

Good job!

Commit:
Updates properties on ChargeState

Updates eu_vehicle and motorized_charge_port, both of these properties
are now optional.

JagCesar reviewed (approved):
LGTM! :D
